daisi/src/minhton-ns3/scenario_steps.h

```diff
@@ -0,0 +1,207 @@
+// Copyright 2023 The SOLA authors
+//
+// This file is part of DAISI.
+//
+// DAISI is free software: you can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation; version 2.
+//
+// DAISI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+// the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with DAISI. If not, see
+// <https://www.gnu.org/licenses/>.
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+#ifndef DAISI_MINHTON_NS3_MINHTON_SCENARIO_STEPS_H_
+#define DAISI_MINHTON_NS3_MINHTON_SCENARIO_STEPS_H_
+
+#include <cstdint>
+#include <optional>
+#include <string>
+
+#include "manager/scenariofile_component.h"
+
+#define TYPE_NAME(name) \
+  static std::string typeName() { return name; }
```

Comment:
Is there a particular reason why there is a macro `TYPE_NAME` here, but e.g. not in `daisi/src/minhton-ns3/peer_discovery_attributes.h`?

Reply:
Not in particular. I now changed everything to directly implement ``typeName()`` and removed the macro.